src/test/java/com/puppycrawl/tools/checkstyle/internal/XpathRegressionTest.java

```diff
@@ -0,0 +1,208 @@
+            "AvoidInlineConditionals",
+            "AvoidNestedBlocks",
+            "BooleanExpressionComplexity",
+            "CatchParameterName",
+            "ClassDataAbstractionCoupling",
+            "ClassFanOutComplexity",
+            "ClassTypeParameterName",
+            "ConstantName",
+            "CovariantEquals",
+            "DescendantToken",
+            "DesignForExtension",
+            "EmptyBlock",
+            "EmptyStatement",
+            "EqualsAvoidNull",
+            "EqualsHashCode",
+            "ExecutableStatementCount",
+            "FinalLocalVariable",
+            "FinalParameters",
+            "HideUtilityClassConstructor",
+            "IllegalInstantiation",
+            "IllegalToken",
+            "IllegalTokenText",
+            "IllegalType",
+            "InnerAssignment",
+            "InnerTypeLast",
+            "InterfaceTypeParameterName",
+            "JavaNCSS",
+            "IllegalImport",
+            "LocalFinalVariableName",
+            "LocalVariableName",
+            "MagicNumber",
+            "MemberName",
+            "MethodLength",
+            "MethodName",
+            "MethodTypeParameterName",
+            "ModifiedControlVariable",
+            "ModifierOrder",
+            "MultipleStringLiterals",
+            "MutableException",
+            "OperatorWrap",
+            "PackageName",
+            "ParameterAssignment",
+            "ParameterName",
+            "ParameterNumber",
+            "RedundantImport",
+            "RedundantModifier",
+            "ReturnCount",
+            "SeparatorWrap",
+            "SimplifyBooleanExpression",
+            "SimplifyBooleanReturn",
+            "StaticVariableName",
+            "StringLiteralEquality",
+            "SuperClone",
+            "SuperFinalize",
+            "SuppressWarnings",
+            "ThrowsCount",
+            "TypeName",
+            "VisibilityModifier"
+    ));
+
+    private static Set<String> simpleCheckNames;
+    private static Map<String, String> allowedDirectoryAndChecks;
+
+    private Path javaDir;
+    private Path inputDir;
+
+    @BeforeClass
+    public static void setUpBeforeClass() throws IOException {
+        simpleCheckNames = CheckUtil.getSimpleNames(CheckUtil.getCheckstyleChecks());
+
+        allowedDirectoryAndChecks = simpleCheckNames
+                .stream()
+                .collect(Collectors.toMap(id -> id.toLowerCase(Locale.ENGLISH), id -> id));
+    }
+
+    @Before
+    public void setUp() throws Exception {
+        javaDir = Paths.get("src/it/java/" + getPackageLocation());
+        inputDir = Paths.get(getPath(""));
+    }
+
+    @Override
+    protected String getPackageLocation() {
+        return "org/checkstyle/suppressionxpathfilter";
+    }
+
+    @Override
+    protected String getResourceLocation() {
+        return "it";
+    }
+
+    @Test
+    public void validateJavaFiles() throws Exception {
```

Comment:
`validateIntegrationTestClassNames` I think would be better.

Reply:
done